src/storage/src/controller.rs

```diff
@@ -2038,7 +2040,7 @@ mod persist_write_handles {
                                                 updates.extend(update);
                                                 old_upper.join_assign(&Antichain::from_elem(upper));
                                             }
-                                            all_responses.push(response);
+                                            all_responses.push((ids, response));
```

Comment:
probably overly nitpicky so I'll leave up to you:  You could make `all_responses` a map of `Sender<..>` -> HashSet<GlobalId>` -- then it'll be a little more parallel with `all_updates`